### PR TITLE
[MAINTENANCE] Making tests for column names insensitivity for Oracle, DB2, and Snowflake easier to read

### PR DESCRIPTION
**Pull request by alexsherstinsky**
### Scope
* Adding `"id"` parameter to test parametrization.
* Moving fixtures closer to where they are used.
* Using descriptive names for test columns.
* Adding docstring for subtle aspect of one test.

### Remarks
* JIRA: DX-307

- [ ] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [x] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [x] Code is linted

    ```
    black .

    ruff . --fix
    ```
- [x] Appropriate tests and docs have been updated

For more details, see our [Contribution Checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist), [Coding style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style), and [Documentation style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/docs_style).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: tests/expectations/metrics/test_metrics_util.py [hunk 1]
```diff
@@ -315,18 +315,24 @@ def test_get_dbms_compatible_metric_domain_column_kwargs_column_not_found(
...
-        ],
-        [
-            "travel_agents",
-            sqlalchemy.quoted_name(value="travel_agents", quote=True),
+            id="column_does_not_need_to_be_quoted",
+        ),
+        pytest.param(
+            "should_be_quoted",
+            sqlalchemy.quoted_name(value="should_be_quoted", quote=True),
             "TRAVEL_AGENTS",
```
Comment (anthonyburdi) [inline tests/expectations/metrics/test_metrics_util.py]: How about this?
```suggestion
            "SHOULD_BE_QUOTED",
```

File: tests/expectations/metrics/test_metrics_util.py [hunk 2]
```diff
@@ -354,20 +360,27 @@ def test_get_dbms_compatible_metric_domain_column_kwargs(
...
     [
-        [
+        "input_column_name_a",
+        "input_column_name_b",
+        "output_column_name_a",
+        "output_column_name_b",
+    ],
+    [
+        pytest.param(
             "ARTISTS",
```
Comment (anthonyburdi) [inline tests/expectations/metrics/test_metrics_util.py]: Could even change this to something like
```suggestion
            "NOT_QUOTED",
```

**Review by anthonyburdi — APPROVED** [review]
LGTM, thank you! Just a few minor suggestions but approving in advance 🙇

File: tests/expectations/metrics/test_metrics_util.py (hunk 1 above)
Comment (alexsherstinsky) [reply tests/expectations/metrics/test_metrics_util.py]: @anthonyburdi `TRAVEL_AGENTS` is not to be quoted.  So I will rename accordingly.  Thanks.

File: tests/expectations/metrics/test_metrics_util.py (hunk 2 above)
Comment (alexsherstinsky) [reply tests/expectations/metrics/test_metrics_util.py]: @anthonyburdi Fixed -- committed!  Thanks!

**Pull request merged**